### PR TITLE
Separate QML objects between application and core lib.

### PR DESCRIPTION
**Pull request by rdumusc**
Touch buttons only on Master application.

### CODE REVIEW

File: apps/DisplayCluster/resources/ContentWindow.qml [hunk 1]
```diff
@@ -35,13 +36,27 @@ Rectangle {
...
-            anchors.horizontalCenter: parent.horizontalCenter
-            anchors.bottom: parent.bottom
-            anchors.bottomMargin: 30
-            onClose: contentWindowTouchArea.close()
-            onToggleFullscreen: controller.toggleFullscreen()
             opacity: 0.1
             visible: contentwindow.controlsOpacity > 0 && contentwindow.label != "Dock"
+
+            function buttonTapped(buttonId) {
+                if(buttonId == "close")
```
Comment (tribal-tec) [inline apps/DisplayCluster/resources/ContentWindow.qml]: where does this come from? Comparing a 'magic string'  seems fishy

**Comment by tribal-tec** [comment]
Which coding style we apply for qml? Right now, it's someway different from C++, we makes me a bit unhappy + it's also harder to adapt while reading. Javascript should allow the same style for braces, spaces, etc. like we do in C++. Opinions?

**Comment by rdumusc** [comment]
I was following the coding style that Qt was doing, which is closer to Java style, but I don't have a strong opinion. We could use our C++ style as well. Shall we ask more people / make a poll?

File: apps/DisplayCluster/resources/ContentWindow.qml (hunk 1 above)
Comment (rdumusc) [inline apps/DisplayCluster/resources/ContentWindow.qml]: Ok, I will see if I can find a different way to do that

**Comment by tribal-tec** [comment]
Yes, I'll ask on the mailing list.

**Comment by rdumusc** [comment]
Updated

File: apps/DisplayCluster/resources/ContentWindow.qml [hunk 2]
```diff
@@ -27,21 +27,42 @@ Rectangle {
...
         WindowControls {
-            anchors.horizontalCenter: parent.horizontalCenter
-            anchors.bottom: parent.bottom
-            anchors.bottomMargin: 30
-            onClose: contentWindowTouchArea.close()
-            onToggleFullscreen: controller.toggleFullscreen()
             opacity: 0.1
             visible: contentwindow.controlsOpacity > 0 && contentwindow.label != "Dock"
+
+            // controller is not set yet at Component.onCompleted, so
```
Comment (tribal-tec) [inline apps/DisplayCluster/resources/ContentWindow.qml]: Shouldn't this be fixed then? The function now compensates for that fact? Will be the same for similar new features we add I guess

File: src/core/DisplayGroup.cpp [hunk 3]
```diff
@@ -74,6 +74,11 @@ void DisplayGroup::addContentWindow( ContentWindowPtr contentWindow )
     sendDisplayGroup();
 }
 
+void DisplayGroup::removeContentWindow( const QUuid id )
```
Comment (tribal-tec) [inline src/core/DisplayGroup.cpp]: No const ref?

File: src/core/DisplayGroup.cpp [hunk 4]
```diff
@@ -91,8 +96,16 @@ void DisplayGroup::removeContentWindow( ContentWindowPtr contentWindow )
     sendDisplayGroup();
 }
 
+void DisplayGroup::moveContentWindowToFront( const QUuid id )
```
Comment (tribal-tec) [inline src/core/DisplayGroup.cpp]: dito?

File: src/core/DisplayGroup.cpp [hunk 5]
```diff
@@ -91,8 +96,16 @@ void DisplayGroup::removeContentWindow( ContentWindowPtr contentWindow )
...
+{
+    moveContentWindowToFront( getContentWindow( id ));
+}
+
 void DisplayGroup::moveContentWindowToFront( ContentWindowPtr contentWindow )
 {
+    if( contentWindow == contentWindows_.back( ))
```
Comment (tribal-tec) [inline src/core/DisplayGroup.cpp]: windows at front are back in the list? :)

Comment (rdumusc) [inline src/core/DisplayGroup.cpp]: yes, they are rendered in order, so back is front ;-)

File: src/core/DisplayGroup.cpp (hunk 3 above)
Comment (rdumusc) [inline src/core/DisplayGroup.cpp]: it's a slot, so no const ref

File: src/core/DisplayGroup.cpp (hunk 4 above)
Comment (rdumusc) [inline src/core/DisplayGroup.cpp]: same

File: apps/DisplayCluster/resources/ContentWindow.qml (hunk 2 above)
Comment (rdumusc) [inline apps/DisplayCluster/resources/ContentWindow.qml]: working on it, creating the controller outside of the touchArea.

**Comment by rdumusc** [comment]
see last update for controller in QML

**Comment by tribal-tec** [comment]
+2. good to merge!

**Comment by rdumusc** [comment]
retest this please

**Pull request merged**